util/argo/argo.go

```diff
@@ -432,7 +431,6 @@ func validateRepo(ctx context.Context,
 		permittedHelmCredentials,
 		enabledSourceTypes,
 		settingsMgr,
-		app.Spec.HasMultipleSources(),
```

Comment:
Can you share your reasoning here? I think we only include those specific parameters for a reason. I don't think we should send the whole object.

Reply:
Hi! Passing a complex structure as an argument rather than a long list of individual arguments offers several advantages in terms of clarity, maintainability, and flexibility of the code.

In this case we are using Application instead to different individual Application fields

I haven't strong opinion on that then I can undo this change if you think is better

Reply:
I think we're good. I'll resolve this. The object is staying in repo server so it's ok. We have to be careful what we pass to manifest generation but this should be fine.